client/src/components/addName.js

```diff
@@ -70,18 +70,19 @@ function AddName() {
         //unique constraint error(name already exists)
         if (response.status === 422) {
           setUniqueError(true);
+          return Promise.reject(response);
         }
         return response.json();
       })
       .then((data) => {
         //Name successfully submitted to the database:
         if (!uniqueError) {
-          setAddedName(data.name);
-          modalRef.current.showModal();
+          setAddedName(data[0].name);
         }
       })
       .catch((error) => console.log(error))
       .finally(() => {
+        modalRef.current.showModal();
```

Comment:
We want to show the modal regardless of whether promises resolve or reject, so moving it to the `finally()` block.

Reply:
I like it!  Thanks for cleaning up my code.